### DISCUSSION TITLE
blog/2020/07/25/capture_screen_using_opencv

### DISCUSSION THREAD

**Discussion by giscus** [SuperKogito/SuperKogito.github.io#8] [Announcements]
# blog/2020/07/25/capture_screen_using_opencv

Capturing the screen using OpenCV

https://superkogito.github.io/blog/2020/07/25/capture_screen_using_opencv.html

**Comment by barakdiker123** [comment]
Will this work on Mac or Linux ?

**Comment by SuperKogito** [comment]
I doubt that it will work as it is, since it has some Windows dependencies as shown in the first code section. 
```cpp
#pragma once
#include <Windows.h>
#include <opencv2/opencv.hpp>

using nammespace cv;
```

Most probably, you will need to edit those dependencies to match your platform.

**Comment by JMC-CR** [comment]
Hi any idea why buf.clear() is undefined?

**Comment by SuperKogito** [comment]
Idk why you get this error since .clear() is definitely part of the vector class https://en.cppreference.com/w/cpp/container/vector/clear
make sure that you are defining your vector variable above the clearing and it is not inside some sort of a loop/if condition so it can be seen
